### ISSUE TITLE
Infinite loop for `Lint/SafeNavigationChain` and `Lint/SafeNavigationConsistency`

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#13341]
```ruby
foo&.bar&.baz && foo&.bar&.zoo
```

```
Offenses:

example.rb:3:26: W: [Corrected] Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.
foo&.bar&.baz && foo&.bar.zoo
                         ^^^^
example.rb:3:26: W: [Corrected] Lint/SafeNavigationConsistency: Use . instead of unnecessary &..
foo&.bar&.baz && foo&.bar&.zoo
                         ^^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in example.rb and caused by Lint/SafeNavigationConsistency -> Lint/SafeNavigationChain
Hint: Please update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
Please check the latest version at https://rubygems.org/gems/rubocop.
```
